src/prpy/util.py

```diff
@@ -693,3 +693,44 @@ def FindCatkinResource(package, relative_path):
     else:
         raise IOError('Loading resource "{:s}" failed.'.format(
                       relative_path))
+
+
+def IsAtTrajectoryStart(robot, trajectory):
+    """
+    Check if robot's active DOFs match the start configuration of a trajectory.
+
+    Examines the current active DOF values of the specified robot and compares
+    these values to the first waypoint of the specified trajectory.  If every
+    DOF value differs by less than the DOF resolution of the specified
+    joint/axis, then the function returns True.  Otherwise, it returns False.
+
+    @param robot: the robot whose active DOFs will be checked
+    @param trajectory: the trajectory whose start configuration will be checked
+    @returns: True if the robot's active DOFs match the given trajectory
+              False if one or more active DOFs differ by DOF resolution
+    """
+    # Get current active configuration of robot.
+    dof_indices = robot.GetActiveDOFIndices()
+    dof_values = robot.GetActiveDOFValues()
+    dof_resolutions = robot.GetActiveDOFResolutions()
```

Comment:
Ok, but what *should* it do?  Keep in mind that we are under no obligation to use OpenRAVE defaults here.  ExecuteTrajectory can splice the trajectory however we see fit before passing it into the SetPath function.

It seems like completely ignoring ActiveDOFs is not particularly intuitive behavior.  Doesn't it?